src/main/java/net/sf/jsqlparser/statement/StatementImpl.java

```diff
@@ -0,0 +1,46 @@
+/*-
+ * #%L
+ * JSQLParser library
+ * %%
+ * Copyright (C) 2004 - 2021 JSQLParser
+ * %%
+ * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
+ * #L%
+ */
+package net.sf.jsqlparser.statement;
+
+/**
+ *
+ * @author <a href="mailto:andreas@manticore-projects.com">Andreas Reichel</a>
+ */
+public abstract class StatementImpl implements Statement {
+
+    public boolean isQuery() {
+        return false;
+    }
+
+    public boolean isDML() {
```

Comment:
The was pure convenience for the end user:

1. the method `getStatementType()` is useful for `switch case` statements
2. the methods `isFoo()` are useful for `if else` statements.

Example: Imagine a SQL Editor as part of a Web Application, where certain normal users shall be able to execute queries but only Admin users shall be able to execute DDLs and maybe only privileged would run DMLs. In this scenario these methods would become very handy because otherwise you will end-up with a **long** list like:
```java 
if (foo instanceof Select) {
     Select select = (Select) foo;
     ...
} else (foo instanceof Update) {
    ...
} ...
```